plugins/controlboard/src/ControlBoardDriver.cpp

```diff
@@ -1144,13 +1159,120 @@ bool GazeboYarpControlBoardDriver::setPIDsForGroup_IMPEDANCE(std::vector<std::st
     }
 }
 
-bool GazeboYarpControlBoardDriver::setMinMaxImpedance()
+bool GazeboYarpControlBoardDriver::FindMotorControlGroup(yarp::os::Bottle& motorControlGroup_bot) const
 {
+    if (!m_pluginParameters.check("WRAPPER"))
+    {
+        yError()<<"Missing WRAPPER group";
+        return false;
+    }
+
+    if (!m_pluginParameters.findGroup("WRAPPER").check("networks"))
+    {
+        yError()<<"Missing networks group";
+        return false;
+    }
 
     yarp::os::Bottle& name_bot = m_pluginParameters.findGroup("WRAPPER").findGroup("networks");
     std::string name = name_bot.get(1).toString();
 
-    yarp::os::Bottle& kin_chain_bot = m_pluginParameters.findGroup(name);
+    motorControlGroup_bot = m_pluginParameters.findGroup(name);
+    return true;
+}
+
+bool GazeboYarpControlBoardDriver::setPositionsToleranceLinear()
+{
+    yarp::os::Bottle kin_chain_bot;
+    if (!FindMotorControlGroup(kin_chain_bot))
+        return false;
+
+    if (!kin_chain_bot.check("positionToleranceLinear")) {
+        yWarning()<<"No positionToleranceLinear value found in ini file, default one will be used!";
+        yDebug()<<"positionToleranceLinear: [ "<<m_robotPositionToleranceLinear<<" ]";
```

Comment:
If we intend to leave this in the code, I would turn it in a warning for consistency with the previous message, otherwise I would remove it.

Reply:
I remove